packages/ovm/src/app/utils.ts

```diff
@@ -61,34 +62,64 @@ export interface OvmTransactionMetadata {
 }
 
 /**
- * Convert internal logs into OVM logs. Or in other words, take the logs which
+ * Convert internal transaction logs into OVM logs. Or in other words, take the logs which
  * are emitted by a normal Ganache or Geth node (this will include logs from the ExecutionManager),
  * parse them, and then convert them into logs which look like they would if you were running this tx
  * using an OVM backend.
  *
+ * NOTE: The input logs MUST NOT be stripped of any Execution Manager events, or this function will break.
  *
- * @param logs an array of internal logs which we will parse and then convert.
+ * @param logs an array of internal transaction logs which we will parse and then convert.
  * @return the converted logs
  */
-export const convertInternalLogsToOvmLogs = (logs: Log[]): Log[] => {
+export const convertInternalLogsToOvmLogs = (
+  logs: Log[],
+  executionManagerAddress: string
+): Log[] => {
   let activeContract = logs[0] ? logs[0].address : ZERO_ADDRESS
   const loggerLogs = [`Parsing logs from contract ${activeContract}: `]
   const ovmLogs = []
+  let cumulativeTxEMLogIndices = 0
+  let prevEMLogIndex = 0
   logs.forEach((log) => {
-    const executionManagerLog = executionManagerInterface.parseLog(log)
-    if (executionManagerLog) {
-      if (executionManagerLog.name === 'ActiveContract') {
-        activeContract = executionManagerLog.values['_activeContract']
+    if (log.address.toUpperCase() === executionManagerAddress.toUpperCase()) {
+      const EMLogIndex = log.logIndex
+      if (EMLogIndex < prevEMLogIndex) {
```

Comment:
should tihs be `<=`? E.g. if prevEMlogIndex is 0, and EMLogIndex is 0, it's still a new transaction, right?

Reply:
nice catch!